src/sdam/state/server.rs

```diff
@@ -74,7 +74,10 @@ impl Server {
 /// TODO: add success cases from application handshakes.
 #[derive(Debug)]
 pub(crate) enum ServerUpdate {
-    Error { error: Error, error_generation: u32 },
+    Error {
+        error: String,
```

Comment:
Passed an owned error over the channel is no longer possible since the error also needs to be returned to the user in some places (e.g. when a new connection is created but fails to establish during a checkout). Luckily, the other side of the channel only uses the errors to mark the server as unknown in the topology, and an earlier commit in this PR already updated ServerDescription to use String instead of Error, so we can just pass the string across the channel eagerly instead of lazily getting the message on the other end.